src/transform.py

```diff
@@ -45,25 +50,221 @@ def extract(cls, values):
         return values
 
 
+class PretalxSlot(BaseModel):
+    room: str | None = None
+    start: datetime | None = None
+    end: datetime | None = None
+
+    @field_validator("room", mode="before")
+    @classmethod
+    def handle_localized(cls, v):
+        if isinstance(v, dict):
+            return v.get("en")
+        return v
+
+
+class TimingRelationship(BaseModel):
+    talks_in_parallel: list[str]
+    talks_after: list[str]
+    talks_before: list[str]
+    next_talk: str | None = None
+    prev_talk: str | None = None
+
+    @model_validator(mode="before")
+    @classmethod
+    def compute(cls, values):
+        session = values["session"]
+        all_sessions = values["all_sessions"]
+
+        talks_in_parallel = cls.compute_talks_in_parallel(session, all_sessions)
+        talks_after_data = cls.compute_talks_after(
+            session, all_sessions, talks_in_parallel
+        )
+        talks_before_data = cls.compute_talks_before(
+            session, all_sessions, talks_in_parallel
+        )
+
+        values["talks_in_parallel"] = talks_in_parallel
+        values["talks_after"] = talks_after_data.get("talks_after")
+        values["talks_before"] = talks_before_data.get("talks_before")
+        values["next_talk"] = talks_after_data.get("next_talk")
+        values["prev_talk"] = talks_before_data.get("prev_talk")
+
+        return values
+
+    @staticmethod
+    def compute_talks_in_parallel(
+        session: PretalxSession, all_sessions: list[PretalxSession]
+    ) -> list[str]:
+        talks_parallel = []
+        for other_session in all_sessions:
+            if (
+                other_session.code == session.code
+                or other_session.start is None
+                or session.start is None
+            ):
+                continue
+
+            # If they intersect, they are in parallel
+            if other_session.start < session.end and other_session.end > session.start:
+                talks_parallel.append(other_session.code)
+
+        return talks_parallel
+
+    @staticmethod
+    def compute_talks_after(
```

Comment:
This one looks a bit complex, we could consider refactoring it a bit after introducing some tests.